File: BE/Src/MIT.Fwk.Infrastructure/Entities/Tenant.cs
using JsonApiDotNetCore.Resources;
using JsonApiDotNetCore.Resources.Annotations;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace MIT.Fwk.Infrastructure.Entities
{

    [Table("Tenants")]
    public class Tenant : Identifiable<int>
    {

        [Attr]
        public string Name { get; set; }

        [Attr]
        public string Description { get; set; }

        [Attr]
        public string Organization { get; set; }

        [Attr]
        public bool Enabled { get; set; }

        [Attr]
        public int ParentTenant { get; set; }

        [Attr]
        public string TenantVAT { get; set; }

        [Attr]
        public string TaxId { get; set; }

        [Attr]
        public string Email { get; set; }

        [Attr]
        public string TenantPEC { get; set; }

        [Attr]
        public string PhoneNumber { get; set; }

        [Attr]
        public string WebSite { get; set; }

        [Attr]
        public string TenantSDI { get; set; }

        [Attr]
        public string TenantIBAN { get; set; }

        [Attr]
        public string Owner { get; set; }

        [Attr]
        public string Commercial { get; set; }

        [Attr]
        public decimal? ShareCapital { get; set; }

        [Attr]
        public string RegisteredOfficeAddress { get; set; }

        [Attr]
        public string RegisteredOfficeCity { get; set; }

        [Attr]
        public string RegisteredOfficeProvince { get; set; }

        [Attr]
        public string RegisteredOfficeState { get; set; }

        [Attr]
        public string RegisteredOfficeRegion { get; set; }

        [Attr]
        public string RegisteredOfficeZIP { get; set; }

        [Attr]
        public string BillingAddressAddress { get; set; }

        [Attr]
        public string BillingAddressCity { get; set; }

        [Attr]
        public string BillingAddressProvince { get; set; }

        [Attr]
        public string BillingAddressState { get; set; }

        [Attr]
        public string BillingAddressRegion { get; set; }

        [Attr]
        public string BillingAddressZIP { get; set; }

        [Attr]
        public bool matchBillingAddress { get; set; }

        [Attr]
        public bool isErasable { get; set; }

        [Attr]
        public bool isRecovery { get; set; }

        [Attr]
        public string FreeFieldString1 { get; set; }
        [Attr]
        public string FreeFieldString2 { get; set; }
        [Attr]
        public string FreeFieldString3 { get; set; }
        [Attr]
        public DateTime? FreeFieldDateTime1 { get; set; }
        [Attr]
        public DateTime? FreeFieldDateTime2 { get; set; }
        [Attr]
        public int? FreeFieldInt2 { get; set; }
        [Attr]
        public int? FreeFieldInt1 { get; set; }
        [Attr]
        public decimal? FreeFieldFloat { get; set; }
        [Attr]
        public bool? FreeFieldBoolean1 { get; set; }
        [Attr]
        public bool? FreeFieldBoolean2 { get; set; }

        [HasMany]
        public virtual ICollection<UserRole> UserRoles { get; set; }

        [HasMany]
        public virtual ICollection<UserTenant> UserTenants { get; set; }

        [HasMany]
        public virtual ICollection<Category> Categories { get; set; }

        [HasMany]
        public virtual ICollection<BlockNotification> BlockNotifications { get; set; }
    }
}

File: FE/WEB/app/models/tenant.js
import Model, { attr, hasMany } from '@ember-data/model';

export default class TenantModel extends Model {
  @hasMany('user-tenant', { async: true, inverse: 'tenant' }) userTenant;
  @hasMany('category', { async: true, inverse: 'tenant' }) category;
  @hasMany('media-category', { async: true, inverse: 'tenant' }) mediaCategory;

  @attr('string', { defaultValue: '' }) name;
  @attr('number', { defaultValue: 0 }) parentTenant;
  @attr('boolean', { defaultValue: false }) enabled;
  @attr('boolean', { defaultValue: true }) isErasable;
  @attr('boolean', { defaultValue: false }) isRecovery;
  @attr('string', { defaultValue: '' }) description;

  @attr('string', { defaultValue: '' }) organization;

  @attr('string', { defaultValue: '' }) tenantVAT;
  @attr('string', { defaultValue: '' }) taxId;
  @attr('string', { defaultValue: '' }) email;
  @attr('string', { defaultValue: '' }) tenantPEC;
  @attr('string', { defaultValue: '' }) phoneNumber;
  @attr('string', { defaultValue: '' }) webSite;
  @attr('string', { defaultValue: '' }) tenantSDI;
  @attr('string', { defaultValue: '' }) tenantIBAN;
  @attr('string', { defaultValue: '' }) owner;
  @attr('string', { defaultValue: '' }) commercial;
  @attr('number', { defaultValue: 0 }) shareCapital;
  @attr('string', { defaultValue: '' }) registeredOfficeAddress; // indirizzo della sede legale
  @attr('string', { defaultValue: '' }) registeredOfficeCity;
  @attr('string', { defaultValue: '' }) registeredOfficeProvince;
  @attr('string', { defaultValue: '' }) registeredOfficeState;
  @attr('string', { defaultValue: '' }) registeredOfficeRegion;
  @attr('string', { defaultValue: '' }) registeredOfficeZIP;

  @attr('boolean', { defaultValue: true }) matchBillingAddress;
  @attr('string', { defaultValue: '' }) billingAddressAddress;
  @attr('string', { defaultValue: '' }) billingAddressCity;
  @attr('string', { defaultValue: '' }) billingAddressProvince;
  @attr('string', { defaultValue: '' }) billingAddressState;
  @attr('string', { defaultValue: '' }) billingAddressRegion;
  @attr('string', { defaultValue: '' }) billingAddressZIP;

  // free-field
  @attr('string', { defaultValue: '' }) freeFieldString1;
  @attr('string', { defaultValue: '' }) freeFieldString2;
  @attr('string', { defaultValue: '' }) freeFieldString3;
  @attr('number', { defaultValue: 0 }) freeFieldInt1;
  @attr('number', { defaultValue: 0 }) freeFieldInt2;
  @attr('number', { defaultValue: 0 }) freeFieldFloat; // dimensione: (10,3)
  @attr('boolean', { defaultValue: false }) freeFieldBoolean1;
  @attr('boolean', { defaultValue: false }) freeFieldBoolean2;
  @attr('date', {
    defaultValue() {
      return new Date();
    },
  })
  freeFieldDateTime1;
  @attr('date', {
    defaultValue() {
      return new Date();
    },
  })
  freeFieldDateTime2;
}
